File: app/src/main/java/com/example/willbe/flowerpassword/ui/SettingsActivity.java
package com.example.willbe.flowerpassword.ui;

import android.app.FragmentManager;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.ActionBarActivity;

import com.example.willbe.flowerpassword.R;

/**
 * Created by xpf on 2015/4/19.
 */
public class SettingsActivity extends ActionBarActivity {


    private static final String SHOW_PASS = "show_pass";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_preference);
        FragmentManager manager = getFragmentManager();
        manager.beginTransaction().replace(R.id.container, new PreferenceFragment()).commit();
    }


    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    public static boolean isShowAfterEncryption(Context context) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getBoolean(SHOW_PASS, true);

    }
}

File: app/src/main/kotlin/SettingsActivity.kt
package com.example.willbe.flowerpassword

import android.content.Context
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import com.example.willbe.flowerpassword.R

/**
 * Created by xpf on 2015/4/19.
 */
public class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_preference)
        val manager = getFragmentManager()
        manager.beginTransaction().replace(R.id.container, PreferenceFragment()).commit()
    }


    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    companion object {


        private fun SHOW_PASS() = "show_pass"

        public fun isShowAfterEncryption(context: Context): Boolean {
            val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
            return sharedPreferences.getBoolean(SHOW_PASS(), true)

        }
    }
}